docs/proposals/20221611-workload-identity-integration.md

```diff
@@ -0,0 +1,418 @@
+		* [2. How to distribute key pair to management cluster?](#Howtodistributekeys)
+		* [3. User Experience](#UserExperience)
+	* [Limitations](#Limitations)
+	* [Migration Plan](#MigrationPlan)
+	* [Test Plan](#TestPlan)
+* [Implementation History](#ImplementationHistory)
+
+<!-- vscode-markdown-toc-config
+	numbering=false
+	autoSave=false
+	/vscode-markdown-toc-config -->
+<!-- /vscode-markdown-toc -->
+
+## <a name='Acronyms'></a>Acronyms
+| Acronym      | Full Form               |
+| ------------ | ------------------------|
+| AD           | Active Directory        |
+| AAD          | Azure Active Directory  |
+| AZWI         | Azure Workload Identity |
+| OIDC         | OpenID Connect          |
+| JWKS         | JSON Web Key Sets       |
+
+## <a name='Summary'></a>Summary
+
+Workloads deployed in Kubernetes cluster may require Azure Active Directory application credential or managed identities to access azure protected resource e.g. Azure Key Vault, Virtual Machines etc. Azure AD Pod Identity helps access azure resources without the need of a secret management via Azure Managed Identities. Azure AD Pod Identity is now deprecated and Azure AD Workload Identity is the next iteration of the former. This design proposal aims to define the way for AZWI integration into capz for self managed clusters with keeping in mind other factor e.g. User Experience, Multi Tenancy, Backwards Compatibility etc.
+
+For more details about Azure AD Pod Identity please visit this [link](https://github.com/Azure/aad-pod-identity)  
+
+## <a name='Motivation'></a>Motivation
+
+AZWI gives the capability to federate the credentials with external identity providers in a Kubernetes native way. This approach overcomes several limitations of Azure AD Identity as mentioned below.
+- Removes scale and performance issues that existed for identity assignment.
+- Supports K8s clusters hosted in any cloud or on premise.
+- Supports both Linux and Windows workloads and removes the need for CRD and pods that intercept IMDS traffic.
+
+From CAPZ perspective
+- The AAD pod identity pod has to be deployed on all the nodes where the capz pod can be potentially scheduled.
+- It has a dependency with the CNI and requires modifying iptables on the node.
+- AAD pod identity is now deprecated. 
+
+To learn more about AZWI please visit this link https://azure.github.io/azure-workload-identity/docs/introduction.html
+
+### <a name='Goals'></a>Goals
+
+- Enable CAPZ pod to use AZWI on the management cluster to authenticate to Azure to create/update/delete resources as part of workload cluster lifecycle management
+
+### <a name='NonGoals'></a>Non Goals
+
+- Automation for migration from AAD pod identity to workload identity.
+- More flexible way so key distribution so that key pair other than the one present for the exiting cluster via `<cluster-name>-sa` secret can be used. To understand more on this please refer [MigrationPlan](#migration-plan) 
```

Comment:
can we also add a non-goal/future work to install workload identity on workload clusters created by CAPZ

Reply:
and also one for using workload identity for cloud-provider auth once that's supported

Reply:
done